packages/utils/telemetry-utils/src/errorLogging.ts

```diff
@@ -49,56 +58,96 @@ export function extractLogSafeErrorProperties(error: any) {
     return safeProps;
 }
 
-/** Type guard for ILoggingError */
-export const isILoggingError = (x: any): x is ILoggingError => typeof x?.getTelemetryProperties === "function";
+/** Metadata to annotate an error object when annotating or normalizing it */
+export interface FluidErrorAnnotations {
+    /** Telemetry props to log with the error */
+    props?: ITelemetryProperties;
+    /** fluidErrorCode to mention if error isn't already an IFluidErrorBase */
+    errorCodeIfNone?: string;
+}
+
+/** For backwards compatibility with pre-fluidErrorCode valid errors */
+function patchWithErrorCode(
+    legacyError: Omit<IFluidErrorBase, "fluidErrorCode">,
+    errorCode: string = "<error predates fluidErrorCode>",
+): asserts legacyError is IFluidErrorBase {
+    const patchMe: { fluidErrorCode?: string } = legacyError as any;
+    if (patchMe.fluidErrorCode === undefined) {
+        patchMe.fluidErrorCode = errorCode;
+    }
+}
 
 /**
- * Read-Write Logging Error.  Not exported.
- * This type alias includes addTelemetryProperties, and applies to objects even if not instanceof LoggingError\
+ * Normalize the given error yielding a valid Fluid Error
+ * @returns A valid Fluid Error with any provided annotations applied
+ * @param error - The error to normalize
+ * @param annotations - Annotations to apply to the normalized error
  */
-type RwLoggingError = LoggingError;
+export function normalizeError(
+    error: unknown,
+    annotations: FluidErrorAnnotations = {},
+): IFluidErrorBase {
+    // Back-compat, while IFluidErrorBase is rolled out
+    if (isValidLegacyError(error)) {
+        patchWithErrorCode(error, annotations.errorCodeIfNone);
+    }
+
+    if (isFluidError(error)) {
+        // We can simply annotate the error and return it
+        mixinTelemetryPropsWithFluidError(error, annotations.props);
+        return error;
+    }
+
+    // We have to construct a new fluid error, copying safe properties over
+    const { message, stack } = extractLogSafeErrorProperties(error);
+    const fluidError: IFluidErrorBase = {
+        errorType: "",
+        fluidErrorCode: annotations.errorCodeIfNone ?? "none",
+        message,
+        stack: stack ?? new Error("<<generated stack>>").stack,
```

Comment:
I believer there was a function in logger to generate stack. It was a bit more complex than this given that standard does not guarantee that stack property is populated on construction, it may require throwing error object to populate it (at least that's my recollection  based on reading about it some time ago)

Reply:
I'm aware of that helper.  I thought I remembered reading online that all modern browsers set `stack` on Error creation, but now all I can find is [this MDN page](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/Stack) which warns that stack is non-standard and browser compatibility may be dubious, so yeah I'll switch it to that.